### PR TITLE
Support android compilation of ioselectors.

### PR DESCRIPTION
**Pull request by cheatfate**

### CODE REVIEW

**Comment by endragor** [comment]
Getting this error when importing upcoming asyncdispatch:
```
lib\upcoming\asyncdispatch.nim(1524, 17) Error: attempting to call undeclared routine: 'registerSignal'
```

**Comment by cheatfate** [comment]
@endragor i have fixed all compilation errors.

**Comment by endragor** [comment]
Confirmed this compiles and works on an actual Android device (Galaxy S6).
I'm sorry it took long.

File: lib/pure/ioselects/ioselectors_epoll.nim [hunk 1]
```diff
@@ -194,39 +209,53 @@ proc unregister*[T](s: Selector[T], fd: int|SocketHandle) =
...
+      if pkey.events * {Event.Read, Event.Write} != {}:
+        var epv = epoll_event()
+        if epoll_ctl(s.epollFD, EPOLL_CTL_DEL, fdi.cint, addr epv) == -1:
+          raiseOSError(osLastError())
+        dec(s.count)
+      elif Event.Timer in pkey.events:
+        var epv = epoll_event()
+        if epoll_ctl(s.epollFD, EPOLL_CTL_DEL, fdi.cint, addr epv) == -1:
+          raiseOSError(osLastError())
+        discard posix.close(fdi.cint)
```
Comment (dom96) [inline lib/pure/ioselects/ioselectors_epoll.nim]: There is a lot of discards in this code, please consider replacing them with some error checks.

**Pull request merged**